modules/mcc/include/opencv2/mcc/color.hpp

```diff
@@ -0,0 +1,285 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+//
+//                       License Agreement
+//              For Open Source Computer Vision Library
+//
+// Copyright(C) 2020, Huawei Technologies Co.,Ltd. All rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//             http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Longbu Wang <wanglongbu@huawei.com.com>
+//         Jinheng Zhang <zhangjinheng1@huawei.com>
+//         Chenqi Shan <shanchenqi@huawei.com>
+
+#ifndef __OPENCV_MCC_COLOR_HPP__
+#define __OPENCV_MCC_COLOR_HPP__
+
+#include <map>
+#include "opencv2/mcc/colorspace.hpp"
+#include "opencv2/mcc/distance.hpp"
+
+namespace cv
+{
+namespace ccm
+{
+
+/* *\ brief Color defined by color_values and color space
+*/
+
+class Color
+{
+public:
+
+    /* *\ param grays mask of grayscale color
+        *\ param colored mask of colored color
+        *\ param history storage of historical conversion
+    */
+    cv::Mat colors;
```

Comment:
the same as above are requested: get rid of `cv::`, move implementation outside of .hpp files